docs/en/week12/12-1.md

```diff
@@ -0,0 +1,220 @@
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-1/figure4.png" style="background-color:#DCDCDC;"><br>
+<b>Figure 4:</b> Represent source
+</center>
+
+<br/>
+
+Step 2: Score each source word (attention) 
+Take the dot product between target hidden representation $\vect{z_{t+1}}$ and all the embeddings from source sentence. Then use softmax to get a distribution over source tokens (attention).
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-1/figure5.png" style="background-color:#DCDCDC;"><br>
+<b>Figure 5:</b> Score each source word
+</center>
+
+<br/>
+
+Step 3: Combine target hidden with source vector
+Take weighted sum (weights are in the attention score vector) of the source embeddings and combine it with target hidden representation $\vect{z_{t+1}}$. After final transformation $G(\vect{z})$, we get a distribution $\vect{y_{t+1}}$ over the next word.
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-1/figure6.png" style="background-color:#DCDCDC;"><br>
+<b>Figure 6:</b> Combine target hidden with source vector
+</center>
+
+<br/>
+
+Alignment is learnt implicitly in seq2seq model. All tokens can be processed in parallel efficiently with CNNs or Transformers.
+
+### Test NMT
+After we have the translation model, and know how good they are doing on the training set, we will want to test to see how the model will actually generate translations from the source sentences.
+
+Beam Decoding is used to search in the space of $\vect{y}$ according to
+
+$$ {\operatorname{argmax}} \log p(\vect{y} \mid \vect{x};\theta) $$
+
+Each potential choice (the number of words in vocabulary) has a probability score. At every step, beam search selects the top $k$ scoring among all the branches (maintain a queue with $k$ top scoring paths), then expand each of them and retain the top $k$ scoring paths. As illustrated by the example below:
+
+<center>
+<img src="{{site.baseurl}}/images/week12/12-1/figure7.png" style="background-color:#DCDCDC;"><br>
+<b>Figure 7:</b> Beam Search
+</center>
+
+Let $k = 2$
+1. Start from symbol /s, pick the top 2 scoring from all the possible choices (Life, Today)
+2. Continue from Life and Today, expand each of them and calculate the path score back to the start symbol, pick the top 2 scoring path ("Like was" and "Like is") 
+3. So on and so for, keep proceeding until hit the end of the sentence
+4. At the very last step, select the highest scoring path
+
+Beam search is a greedy procedure and it is very effective in practice. There is a trade-off between computational cost and approximation error (the larger the value of $k$, the better the approximation and the higher the computational cost).
+
+Issues of Beam Search:
+Beam search will always select the highest scoring path. Thus, the solution will be biased because of such strategy. It doesn't handle uncertainty well. 
+
+Other decoding methods: sampling, top-k sampling, generative and discriminative reranking 
```

Comment:
Use `$`…

Reply:
I thought top-k is one word. Okay, will fix that.

Reply:
`top-$k$`, yup.